qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -0,0 +1,244 @@
+            free_parameter_names: Names of the parameter that should be left unbound.
+                If None is specified then all parameters will be bound to their
+                calibrated values.
+        """
+        schedule = self.get_schedule(name, qubits, free_parameter_names)
+
+        gate = Gate(name=name, num_qubits=len(qubits), params=schedule.parameters)
+        circ = QuantumCircuit(self._n_qubits)  # Probably a better way of doing this
+        circ.append(gate, qubits)
+        circ.add_calibration(gate, qubits, schedule, params=schedule.parameters)
+
+        return circ
+
+    def get_schedule(self, name: str, qubits: Tuple,
+                     free_parameter_names: List[str] = None) -> Schedule:
+        """
+        Retrieves a schedule from the instructions dictionary.
+
+        Args:
+            name: Name of the schedule to retrive.
+            qubits: qubits for which to get the schedule.
+            free_parameter_names: List of parameter names that will be left unassigned.
+                The parameter assignment is done by querying the pulse parameter table.
+
+        Returns:
+            schedule for the given (name, qubits) key.
+        """
+        schedule = self._instructions[(name, qubits)]
+
+        if not isinstance(schedule, Schedule):
+            schedule = self.get_composite_instruction(schedule)
+
+        schedule = copy.deepcopy(schedule)
+
+        if not free_parameter_names:
+            free_parameter_names = []
+
+        binding_dict = {}
+        for param in schedule.parameters:
+            if param.name not in free_parameter_names:
+                binding_dict[param] = self._get_parameter_value(param)
+
+        schedule.assign_parameters(binding_dict)
+
+        return schedule
+
+    def _get_parameter_value(self, parameter: ParameterExpression) -> Union[float, int, complex]:
+        """
+        Helper method.
+
+        TODO This could be simplified by better integrating with PulseTable
+        """
+        inst_name, channel, name = parameter.name.split('.')
+        qubits = self._get_qubits(channel)
+
+        return list(self._pulse_table.get_parameter(qubits, channel, inst_name, '', 1.0, name).values())[0].value
+
+    def _add_parameter(self, parameter: ParameterExpression, value: Union[float, int, complex]):
+        """
+        Helper method for current implementation.
+
+        TODO This should/could be moved to PulseTable
+
+        Args:
+            parameter: A parameter to add to the DB.
+            value: The value of the parameter to add to the DB.
+        """
+        inst_name, channel, name = parameter.name.split('.')
+        qubits = self._get_qubits(channel)
+
+        # TODO see if this will stay like this
+        pulse_type = ''
+
+        self._pulse_table.set_parameter(qubits, channel, inst_name, pulse_type, 1.0, name, value)
+
+    def _get_qubits(self, channel_name: str) -> Tuple[int]:
```

Comment:
you don't need this logic here. `backend.configuration().channels` provides a dictionary like this.
```
{'acquire0': {'operates': {'qubits': [0]},
  'purpose': 'acquire',
  'type': 'acquire'},
 'acquire1': {'operates': {'qubits': [1]},
  'purpose': 'acquire',
  'type': 'acquire'},
...
 'd0': {'operates': {'qubits': [0]}, 'purpose': 'drive', 'type': 'drive'},
 'd1': {'operates': {'qubits': [1]}, 'purpose': 'drive', 'type': 'drive'},
...
 'm0': {'operates': {'qubits': [0]}, 'purpose': 'measure', 'type': 'measure'},
 'm1': {'operates': {'qubits': [1]}, 'purpose': 'measure', 'type': 'measure'},
...
 'u0': {'operates': {'qubits': [0, 1]},
  'purpose': 'cross-resonance',
  'type': 'control'},
 'u1': {'operates': {'qubits': [1, 0]},
  'purpose': 'cross-resonance',
  'type': 'control'},
...
}
```

You just need to 
```python
tuple(backend.configuration().channels[channel_name]['operates']['qubits'])
```

However preparing this dictionary is kind of overhead for users if they don't have backend object, i.e. simulator or unittest. So I recommend to extract relationship between channel and qubit in the factory method like:

https://github.com/nkanazawa1989/qiskit-ignis/blob/3c258cabaa046e4fc3bae5837a7bf27f680863ca/qiskit/ignis/experiments/calibration/instruction_data/interface.py#L67-L69

then we can directly get qubits with `qubits = self._channel_map[channel_name]`.

Reply:
Good change. Thanks. I updated the PR.